### ISSUE TITLE
Non-admin can't reuse existing file when updating or creating collection item

### ISSUE THREAD

**Issue by spalberg** [directus/app#1284]
# Bug Report
When a user with non-admin-role tries to use an existing image in an image field an error is thrown (see "Actual Behavior"

## Steps to Reproduce

1. Create a collection that contains a file-field.
2. Create a new role and set read, write and update for the created collection in 1. Let everything else on the default values.
3. Create a new user and assign him the created role
4. Log in as the new user and create or update an item of the collection from 1.
5. For the file-field, choose "Existing" and select any prev uploaded image
6. Save the item

## Expected Behavior
Item should be saved

## Actual Behavior
Error is thrown and red toast is shown in the frontend.
Response from the api:
```
{
  "error": {
    "code": 302,
    "message": "Updating item from \"directus_users\" collection was denied", 
     "class": "Directus\\Permissions\\Exception\\ForbiddenCollectionUpdateException",
    "file": "\/var\/app\/directus\/src\/core\/Directus\/Permissions\/Acl.php",
    "line": 1072
  }
}
```

## Other Context & Screenshots

Also reproducible using the lastest docker image and the setup from https://github.com/directus/docker/blob/master/examples/single-api/docker-compose.yml

## Technical Details

- See docker-compose.yml linked above

**Comment by rijkvanzanten** [comment]
I don't think you can, we but do @spalberg 🙂

**Comment by rijkvanzanten** [comment]
The application will have to have a "filter" step between staging the values and saving them, so it has a chance to filter out things like this. In the current state architecture, this is a bit hard to implement.

The application needs the full object in order to render the image preview. It should only stage the ID for saving, but that would mean that we don't have the necessary info to render the preview 🤔 

We could theoretically fetch the relational information in the interface itself asynchronously, but I don't know if that's the cleanest way to go forward 🤔 

thoughts:
* add 'filter' step right before saving to filter out file info, or;
* only stage the ID and fetch file info async in the interface

**Comment by mvhirsch** [comment]
I'm getting the exact same issue, by using the "Files"-interface on my collection. Even after deleting the _modified_by_ column in my collection, I still can't add new files-relations. Is there anything I could try?

**Comment by benhaynes** [comment]
Is this the same issue as M2M @rijkvanzanten ? If so, we should prioritize that relational fix since this seems to be a critical issue. I know it requires a big refactor, but at this point I think we need to resolve it.

**Comment by rijkvanzanten** [comment]
The OP has been fixed in 498436a6. I'll do the same for the files (plural) next.

**Comment by rijkvanzanten** [comment]
The same issue in `files` has been fixed in https://github.com/directus/app/pull/1881
